File: SlimFitGym_Mobile/SlimFitGym_Mobile/Models/TrainingModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SlimFitGym_Mobile.Models
{
    public class TrainingModel
    {
        public int Id { get; set; }
        public int? TrainerId { get; set; }
        public string? Trainer { get; set; }
        public int? RoomId { get; set; }
        public string? Room { get; set; }
        public string Name { get; set; }
        public DateTime TrainingStart { get; set; }
        public DateTime TrainingEnd { get; set; }
        public int MaxPeople { get; set; }
        public bool IsActive { get; set; } = true;
    }
}

File: Frontend/src/app/Models/training.model.ts
export interface TrainingModel {
    id: number
    trainer: string
    trainerId: number
    room: string
    roomId: number
    name: string
    trainingStart: Date
    trainingEnd: Date
    maxPeople: number
    freePlaces: number
    isActive: boolean
    trainerImageUrl: string
    roomImageUrl: string


    //for the display
    userApplied: boolean;
    trainingStartString?: string,
    trainingEndString?: string
  }
  